src/GDGUkraine/rest_controller.py

```diff
@@ -372,77 +373,65 @@ def export_participants(self, id):
-                (lambda x: event.title),
-                (lambda x: x.User.full_name),
-                (lambda x: x.User.gender),
-                (lambda x: x.User.nickname or ''),
-                (lambda x: x.User.email),
-                (lambda x: x.User.phone or ''),
-                (lambda x: x.User.gplus),
-                (lambda x: x.User.hometown or ''),
-                (lambda x: x.User.company or ''),
-                (lambda x: x.User.position or ''),
-                (lambda x: x.User.www or ''),
-                (lambda x: x.User.experience_level or ''),
-                (lambda x: x.User.experience_desc or ''),
-                (lambda x: x.User.interests or ''),
-                (lambda x: x.User.events_visited or ''),
-                (lambda x: x.User.english_knowledge or ''),
-                (lambda x: x.User.t_shirt_size or ''),
-                (lambda x: x.User.additional_info or ''),
-                (lambda x: (
-                    json.dumps(x.EventParticipant.fields)
-                    if x.EventParticipant.fields
-                    else ''
-                )),
-                (lambda x: x.EventParticipant.confirmed),
-            ],
-            headers=[
-                'Registration id',
-                'Registration date',
-                'Event id',
-                'Event title',
-                'Name',
-                'Gender',
-                'Nickname',
-                'Email',
-                'Phone',
-                'Google +',
-                'City',
-                'Company',
-                'Position',
-                'Website',
-                'Experience level',
-                'Experience description',
-                'Interests',
-                'Events visited',
-                'English knowledge',
-                'T-Shirt size',
-                'Additional info',
-                'Extra fields',
-                'Confirmed',
-            ],
-        )
+
+        # Set appropriate headers
+        filename = re.compile(r'[^\w-]').sub('', event.title.replace(' ', '_'))
+
         cherrypy.response.headers['Content-Type'] = (
             'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet'
         )
-        filename = re.compile(r'[^\w-]').sub('', event.title.replace(' ', '_'))
         cherrypy.response.headers['Content-Disposition'] = (
             'attachment; filename={}-{}-{}-participants.xlsx'.format(
                 event.id, filename, event.date,
             )
         )
-        return file_generator(exporter.get_xlsx_content())
+
+        # Retrieve participation data
+        part_data = api.find_participants_by_event(orm_session, event)
+
+        return file_generator(gen_participants_xlsx(part_data))
+
+    @cherrypy.tools.json_out()
+    @cherrypy.tools.authorize()
+    def generate_invites(self, id):
+        return {}
+
+    @cherrypy.tools.json_out()
+    @cherrypy.tools.authorize()
+    def generate_report(self, id, mode=None):
```

Comment:
We cannot do this, since handler has `json_out()` enabled and expects JSON input as well. It won't even catch the request. They are completely different.